doc/_dev-guide/chapter_4/section_3/README.md

```diff
@@ -0,0 +1,445 @@
+```
+
+```shell
+export PATH=${BSDPORTS_PREFIX}/bin:${BSDPORTS_PREFIX}/sbin:${PATH}
+```
+
+```shell
+PKGCONF_PATH=$(pkgconf --variable pc_path pkgconf)
+export PKG_CONFIG_PATH=${PKGCONF_PATH}:$PKG_CONFIG_PATH
+```
+
+```shell
+export CFLAGS="-I${BSDPORTS_PREFIX}/include/ ${CFLAGS}"
+export CXXFLAGS="${CFLAGS} ${CXXFLAGS}"
+export LDFLAGS="-L${BSDPORTS_PREFIX}/lib ${LDFLAGS}"
+```
+
+**Note:** It could also happen that you must provide here further library inlcude and lib paths
+{: .notice}
+
+### Getting the source
+
+To get the latest master from {{ site.product.short_name }} git you can use
+
+```shell
+cd YOUR_PREFERRED_WORKING_DIR   # Replace `YOUR_PREFERRED_WORKING_DIR` with your actual preferred working dir 
+git clone https://github.com/syslog-ng/syslog-ng . 
+```
+
+### Select the compiler
+
+Latest version of {{ site.product.short_name }} [has dropped support of gcc](https://github.com/syslog-ng/syslog-ng/pull/4897), so now the platform default llvm/clang must be used to complie the source.\
+`gcc` still might compile {{ site.product.short_name }} and most of its modules, but there is no guarantie and support of it anymore.
+
+**Hint:** You can always turn off any problematic module via its feature switch.
+{: .notice--info}
+
+To make sure clang is used you can use (optional):
+
+```shell
+export CC=clang
+export CXX=clang++
+```
+
+For gcc (optional):
+
+```shell
+export CC=gcc     # More precisly, the full path of ypur installed gcc compiler
+export CXX=g++    # More precisly, the full path of ypur installed g++ compiler
+```
+
+### Configuration
+
+> **Note:**
+>
+> * for various reasons not all modules can always be configured, built and used on all FreeBSD versions and architectures
+> * for using all the available modules you might have to install further dependencies as mentioned above
+> * for more details, please see the [[actual state of supported features|dev-macos-mod-sup-status]], and the required [dependencies](#dependencies).
+{: .notice}
+
+#### Using autotool
+
+```shell
+# you should use `gmake` instead of `make` on FreeBSD to configure and build syslog-ng with autotools.
+export MAKE=gmake
+
+./autogen.sh
+
+# it is always a good idea keeping as clean the source folder as pissible, so
+# use a dedicated build folder for easier cleanup
+mkdir build; cd build
+
+# Finally, you can start the configuration with enabling or disabling the various modules e.g. like this
+# NOTE: you might want to use the --prefix /full_path_of_your/installdir/ parameters as well, see bellow Warning!
+../configure --enable-extra-warnings --with-ivykis=system --with-systemd-journal=no --disable-java --disable-java-modules
+```
+
+**Warning:** By a good chance, you might want to install the self built instance first to a custom location to prevent overwriting a possibly already existing installed version. In that case pass `--prefix /full_path_of_your/installdir/` to the `configure` script in the above steps.
```

Comment:
This same sentence has been problematic in an earlier comment. Please refer above